compiler/rustc_mir_build/src/build/matches/test.rs

```diff
@@ -649,12 +649,14 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                 }
             }
 
-            // FIXME(#29623): return `Some(1)` when the values are different.
-            (TestKind::Eq { value: test_val, .. }, TestCase::Constant { value: case_val })
-                if test_val == case_val =>
-            {
-                fully_matched = true;
-                Some(TestBranch::Success)
+            (TestKind::Eq { value: test_val, .. }, TestCase::Constant { value: case_val }) => {
+                if test_val == case_val {
+                    fully_matched = true;
+                    Some(TestBranch::Success)
+                } else {
+                    fully_matched = false;
+                    Some(TestBranch::Failure)
+                }
```

Comment:
```rust
fully_matched = test_val == case_val;
Some(if fully_matched { TestBranch::Success } else { TestBranch::Failure })
```

Reply:
I prefer keeping them separate because the fully_matched logic is already not easy to follow across the function